File: sbschoolmanager-api/src/main/java/com/sbschoolmanager/sbschoolmanager_api/model/Contratar.java
package com.sbschoolmanager.sbschoolmanager_api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.time.LocalDate;

@Entity
@Table(name = "CONTRATAR")
@Getter
@Setter
@NoArgsConstructor
public class Contratar {

    @EmbeddedId
    private ContratarId id;

    @ManyToOne
    @MapsId("codAlumno")
    @JoinColumn(name = "Cod_alumno_FK")
    private Alumno alumno;

    @ManyToOne
    @MapsId("codClaseSkate")
    @JoinColumn(name = "Cod_ClaseSkate_FK")
    private ClaseSkate claseSkate;

    @Column(name = "Fecha_Inicio")
    private LocalDate fechaIncio;

    @Column(name = "Fecha_Fin")
    private LocalDate fechaFin;
}

File: sbschoolmanager_app/lib/models/contratar.dart

import "alumno.dart";
import "clase_skate.dart";

class Contratar {
  final Alumno alumno;
  final ClaseSkate claseSkate;
  final String fechaInicio;
  final String fechaFin;

  Contratar({
    required this.alumno,
    required this.claseSkate,
    required this.fechaInicio,
    required this.fechaFin,
  });

  factory Contratar.fromJson(Map<String, dynamic> json){
    return Contratar(
      alumno: Alumno.fromJson(json["alumno"]),
      claseSkate: ClaseSkate.fromJson(json["claseSkate"]),
      fechaInicio: json["fechaIncio"],
      fechaFin: json["fechaFin"],
       );
  }

  Map<String, dynamic> toJson(){
    return{
      "alumno": alumno.toJson(),
      "claseSkate": claseSkate.toJson(),
      "fechaInicio": fechaInicio,
      "fechaFin": fechaFin,
    };
  }
}